extensions/graphite-emitter/src/main/java/io/druid/emitter/graphite/GraphiteEventConverter.java

```diff
@@ -0,0 +1,241 @@
+    this.whiteListDimsMapper = graphiteConfig.getWhiteListDimsMapper() == null ? DEFAULT_DIM_WHITE_LIST_MAPPER : graphiteConfig.getWhiteListDimsMapper();
+    this.ignoreServiceName = graphiteConfig.getIgnoreServiceName();
+    this.ignoreHostname = graphiteConfig.getIgnoreHostname();
+    this.namespacePrefix = graphiteConfig.getNamespacePrefix();
+  }
+
+  public GraphiteEventConverter(
+      @NotNull String namespacePrefix,
+      @NotNull boolean ignoreHostname,
+      @NotNull boolean ignoreServiceName,
+      SortedMap<String, Set<String>> whiteListDimsMapper
+  )
+  {
+    this.whiteListDimsMapper = whiteListDimsMapper == null ? DEFAULT_DIM_WHITE_LIST_MAPPER : whiteListDimsMapper;
+    this.ignoreServiceName = ignoreServiceName;
+    this.ignoreHostname = ignoreHostname;
+    this.namespacePrefix = namespacePrefix;
+  }
+
+  public boolean isIgnoreHostname()
+  {
+    return ignoreHostname;
+  }
+
+  public boolean isIgnoreServiceName()
+  {
+    return ignoreServiceName;
+  }
+
+  public String getNamespacePrefix()
+  {
+    return namespacePrefix;
+  }
+
+
+  /*
+  * The event is considered in white list if and only if it has a prefix key for dims mapping and a valid metric type
+  * */
+  private boolean isInWhiteList(ServiceMetricEvent event)
+  {
+    return getPrefixKey(event.getMetric(), whiteListDimsMapper) != null;
+  }
+
+
+  /**
+   * @param key       the metric name to lookup
+   * @param whiteList
+   *
+   * @return <tt>null</tt> if the key does not match with any of the prefixes keys in @code metricsWhiteList,
+   * or the prefix in @code whiteListDimsMapper
+   */
+  private String getPrefixKey(String key, SortedMap<String, ?> whiteList)
+  {
+    String prefixKey = null;
+    if (whiteList.containsKey(key)) {
+      return key;
+    }
+    SortedMap<String, ?> headMap = whiteList.headMap(key);
+    if (!headMap.isEmpty() && key.startsWith(headMap.lastKey())) {
+      prefixKey = headMap.lastKey();
+    }
+    return prefixKey;
+  }
+
+  /**
+   * Returns a {@link List} of the white-listed dimension's values to send.
+   * The list is order is the same as the order of dimensions {@code whiteListDimsMapper}
+   *
+   * @param event the event for which will filter dimensions
+   *
+   * @return {@link List}  of the filtered dimension values to send or <tt>null<tt/> if the event is not in the white list
+   */
+  private List<String> getOrderedDimValues(ServiceMetricEvent event)
+  {
+    String prefixKey = getPrefixKey(event.getMetric(), whiteListDimsMapper);
+    if (prefixKey == null) {
+      return null;
+    }
+    ImmutableList.Builder<String> outputList = new ImmutableList.Builder();
+    Set<String> dimensions = whiteListDimsMapper.get(prefixKey);
+    if (dimensions == null) {
+      return Collections.emptyList();
+    }
+    for (String dimKey : dimensions) {
+      String dimValue = (String) event.getUserDims().get(dimKey);
+      if (dimValue != null) {
+        outputList.add(dimValue);
```

Comment:
What would be the side effect,if instead of selected dimensions we would have return all the dimensions of the ServiceMetricEvent,in certain fixed order lets say lexicographically sorted ?

I know the name of metrics would have been very long,but AFAIK graphite provide ways to do aggregates using regex pattern,hence if the end user wants to ignore certain dimensions he can easily do so.